src/Build/BackEnd/Components/Communications/NodeProviderOutOfProcBase.cs

```diff
@@ -284,25 +239,49 @@ protected NodeContext GetNode(string msbuildLocation, string commandLineArgs, in
                     CommunicationsUtilities.Trace("Successfully connected to created node {0} which is PID {1}", nodeId, msbuildProcessId);
                     return new NodeContext(nodeId, msbuildProcessId, nodeStream, factory, terminateNode);
                 }
-#else
-                if (WaitForConnectionFromProcess(clientToServerStream, serverToClientStream, msbuildProcessId, hostHandshake, clientHandshake))
-                {
-                    // Connection successful, use this node.
-
-                    clientToServerStream.DisposeLocalCopyOfClientHandle();
-                    serverToClientStream.DisposeLocalCopyOfClientHandle();
 
-                    CommunicationsUtilities.Trace("Successfully connected to created node {0} which is PID {1}", nodeId, msbuildProcessId);
-                    return new NodeContext(nodeId, msbuildProcessId, clientToServerStream, serverToClientStream, factory, terminateNode);
-                }
-#endif
             }
 
             // We were unable to launch a node.
             CommunicationsUtilities.Trace("FAILED TO CONNECT TO A CHILD NODE");
             return null;
         }
 
+        private List<Process> GetPossibleRunningNodes(out string msbuildName, ref string msbuildLocation)
+        {
```

Comment:
The logic in this method really confuses me.  It looks very redundant and I don't understand the rules it has to abide by.  Can you make it more clear or add some comments?

Or is this method just a copy/paste?

Reply:
Yeah, I cut/pasted this from https://github.com/Microsoft/msbuild/pull/2961/files/05edb1b080e0dcbfdea70630cac273f03754719d#diff-e836c7a5cb2f879641ba4e6a3c720a4bL168 with minimal changes because it's not entirely clear to me either.